src/repr/src/bytes.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright 2018 sqlparser-rs contributors. All rights reserved.
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// This file is derived from the sqlparser-rs project, available at
+// https://github.com/andygrove/sqlparser-rs. It was incorporated
+// directly into Materialize on December 21, 2019.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use mz_ore::cast::CastLossy;
+
+/// Struct for postgres compatible size units which is slightly different from
+/// [`bytesize::ByteSize`]. Instead of MiB or GiB and so on, it uses MB, GB for the sizes
+/// with 1024 multiplier.
+/// Valid units are B, KB, MB, GB, TB, PB with multiples of 1024
+/// where 1MB = 1024 KB
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize, Default)]
+pub struct Bytes(pub u64);
+
+impl Bytes {
+    pub const fn b(size: u64) -> Bytes {
+        Bytes(size)
+    }
+
+    pub const fn kb(size: u64) -> Bytes {
+        Bytes(size * BytesUnit::Kb.value())
+    }
+
+    pub const fn mb(size: u64) -> Bytes {
+        Bytes(size * BytesUnit::Mb.value())
+    }
+
+    pub const fn gb(size: u64) -> Bytes {
+        Bytes(size * BytesUnit::Gb.value())
+    }
+
+    pub const fn tb(size: u64) -> Bytes {
+        Bytes(size * BytesUnit::Tb.value())
+    }
+
+    pub const fn pb(size: u64) -> Bytes {
+        Bytes(size * BytesUnit::Pb.value())
+    }
+
+    pub fn as_u64(&self) -> u64 {
+        self.0
+    }
+    /// Copied logic from [`bytesize::ByteSize::to_string`]
+    /// with slight change to have no space between the number and the unit
+    /// to have the same behaviour as Postgres.
+    pub fn to_string(bytes: u64) -> String {
+        let unit = 1024;
+        let unit_base = 6.907755279; // ln 1000
+        let unit_prefix = ['K', 'M', 'G', 'T', 'P', 'E'];
```

Comment:
PostgreSQL seems to use `kB`, not `KB`. Also `E` isn't a thing that PostgreSQL seems to support?

Reply:
The 'E' was copy pasta from ByteSize, removed it. Updated to `kB` everywhere.